### PR TITLE
readme upgrades plus contributing document

### PR DESCRIPTION
**Pull request by justin1121**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## Before submitting for review
+
+Make sure to give some context and overview in the body of your pull request to make it easier for reviewers to understand your changes. Ideally explain why your particular changes were made the way they are.
+
+Importantly, use [keywords](https://help.github.com/en/articles/closing-issues-using-keywords) such as `Closes #<issue-number>` to indicate any issues or other pull requests related to your work.
+
+# Reporting a Bug
+
+Think you've found a bug? Let us know by opening an [issue in our tracker](https://github.com/mortendahl/tf-encrypted/issues) and apply the "bug" label!
```
Comment (jvmancuso) [inline CONTRIBUTING.md]: this is the wrong link

Comment (ianlivingstone) [reply CONTRIBUTING.md]: 💯

File: README.md [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
+## TF Trusted
 
-TF Trusted allows you to run most Tensorflow models inside of an Intel SGX device. It leverages a Tensorflow custom op to send gRPC messages into the Intel SGX device via Asylo where the model is then run by Tensorflow Lite.
+TF Trusted allows you to run most Tensorflow models inside of an [Intel SGX](https://software.intel.com/en-us/sgx) device. It leverages a Tensorflow custom operation to send gRPC messages into the Intel SGX device via [Asylo](https://asylo.dev/) where the model is then run by Tensorflow Lite.
 
-First clone this repo and follow the instructions [here](tf_trusted_custom_op/README.md) to build the required custom operation.
+The goal of this project is to provide an easy to use framework to experiment with running models securely inside secure enclaves. It **shouldn't** be used in production and is provided for research and experimentation only.
 
-##### Pull Asylo Docker Container
+Please see the [contributing](CONTRIBUTING.md) document on how you can help improve the project.
```
Comment (ianlivingstone) [inline README.md]:
```suggestion
We're always looking for contributors, if you're learning about how you can help improve the project, please check out our [contributing guidelines](CONTRIBUTING.md).
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
 
-We're pinned to version v0.3.4 of the docker container for now.
+## Getting Started
 
-```
-$ docker pull gcr.io/asylo-framework/asylo:buildenv-v0.3.4
-```
+First clone this repo and follow the instructions [here](tf_trusted_custom_op/README.md) to build the required custom operation.
```
Comment (ianlivingstone) [inline README.md]:
```suggestion
To get started, clone this repository and then follow the instructions for building the TensorFlow custom operation located [here](tf_trusted_custom_op/README.md).
```

File: README.md [hunk 4]
```diff
@@ -1,20 +1,18 @@
...
+TF Trusted allows you to run most Tensorflow models inside of an [Intel SGX](https://software.intel.com/en-us/sgx) device. It leverages a Tensorflow custom operation to send gRPC messages into the Intel SGX device via [Asylo](https://asylo.dev/) where the model is then run by Tensorflow Lite.
 
-First clone this repo and follow the instructions [here](tf_trusted_custom_op/README.md) to build the required custom operation.
+The goal of this project is to provide an easy to use framework to experiment with running models securely inside secure enclaves. It **shouldn't** be used in production and is provided for research and experimentation only.
 
-##### Pull Asylo Docker Container
+Please see the [contributing](CONTRIBUTING.md) document on how you can help improve the project.
 
-We're pinned to version v0.3.4 of the docker container for now.
+## Getting Started
```
Comment (ianlivingstone) [inline README.md]: What requirements do I need to have before getting started? Git? Tensorflow? Python? Make? etc? We should list those here.

File: README.md [hunk 5]
```diff
@@ -59,15 +55,15 @@ SDK/PSW can be installed with the following instructions:
 
 https://github.com/intel/linux-sgx
 
-##### Run aesmd Service
+#### Run aesmd Service
 
 The aesmd service manages the SGX device.
```
Comment (ianlivingstone) [inline README.md]: What is aesmd and what does it do? It'd be good to lightly document these here.

File: README.md [hunk 6]
```diff
@@ -1,20 +1,18 @@
-### TF Trusted
+## TF Trusted
```
Comment (ianlivingstone) [inline README.md]: In the getting started section, we tell them how to run and build tf trusted locally. However, we don't tell explicitly them it's for local use only, nor, do we differentiate between what happens locally and on an Intel SGX device.

Can you beef up these sections to be more descriptive as to what the user is accomplishing?

File: README.md (hunk 5 above)
Comment (justin1121) [reply README.md]: Can't really find out much about what it exactly does but can say something like `The Architecture Enclave Service Manager (AESM) allows the Intel SGX device to be used by the host operating system. We can start the AESM service with:`

**Pull request merged**